server/odc-service/src/main/java/com/oceanbase/odc/service/connection/FileSystemConnectionTesting.java

```diff
@@ -0,0 +1,153 @@
+                    new ByteArrayInputStream(TMP_TEST_DATA.getBytes(StandardCharsets.UTF_8)), new ObjectMetadata());
+            DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest();
+            deleteObjectsRequest.setBucketName(storageConfig.getBucketName());
+            deleteObjectsRequest.setKeys(Collections.singletonList(objectKey));
+            cloudClient.deleteObjects(deleteObjectsRequest);
+            return ConnectionTestResult.success(config.getType());
+        } catch (CloudException e) {
+            if (e.getCause() instanceof OSSException) {
+                OSSException cause = (OSSException) e.getCause();
+                switch (cause.getErrorCode()) {
+                    case OSSErrorCode.ACCESS_DENIED:
+                        return new ConnectionTestResult(TestResult.akAccessDenied(storageConfig.getAccessKeyId()),
+                                config.getType());
+                    case OSSErrorCode.INVALID_ACCESS_KEY_ID:
+                        return new ConnectionTestResult(TestResult.invalidAccessKeyId(storageConfig.getAccessKeyId()),
+                                config.getType());
+                    case OSSErrorCode.SIGNATURE_DOES_NOT_MATCH:
+                        return new ConnectionTestResult(
+                                TestResult.signatureDoesNotMatch(storageConfig.getAccessKeyId()), config.getType());
+                    case OSSErrorCode.NO_SUCH_BUCKET:
+                        return new ConnectionTestResult(TestResult.bucketNotExist(storageConfig.getBucketName()),
+                                config.getType());
+                    default:
+                        return new ConnectionTestResult(TestResult.unknownError(e), config.getType());
+                }
+            }
+            // TODO:process s3 error message
+            return new ConnectionTestResult(TestResult.unknownError(e), config.getType());
+        }
+    }
+
+    private String getRegion(ConnectionConfig config) {
+        Pattern pattern = Pattern.compile(getEndPointRegex(config.getType()));
+        Matcher matcher = pattern.matcher(config.getHost());
+        if (matcher.find()) {
+            return matcher.group(1);
+        } else {
+            throw new UnExpectedException("Illegal endpoint");
+        }
+    }
+
+    private CloudProvider getCloudProvider(ConnectType type) {
+        switch (type) {
+            case COS:
+                return CloudProvider.TENCENT_CLOUD;
+            case OBS:
+                return CloudProvider.HUAWEI_CLOUD;
+            case S3A:
+                return CloudProvider.AWS;
+            case OSS:
+                return CloudProvider.ALIBABA_CLOUD;
+            default:
+                throw new UnExpectedException();
+        }
+    }
+
+    private String getEndPointRegex(ConnectType type) {
+        switch (type) {
+            case COS:
+                return COS_ENDPOINT_REGEX;
```

Comment:
as you have so many switch case around your code, reconstruct may needed.
let enum's member hold this fields. or put what programmer must implement to enums class to aware if you want to add some type ,you should do like this way